nifi-commons/nifi-properties/src/test/java/org/apache/nifi/util/NiFiPropertiesTest.java

```diff
@@ -167,4 +168,97 @@ private NiFiProperties loadNiFiProperties(final String propsPath, final Map<Stri
         return NiFiProperties.createBasicNiFiProperties(realPath, additionalProperties);
     }
 
-}
+    @Test
+    public void testShouldVerifyValidFormatPortValue() {
+        // Testing with CLUSTER_NODE_PROTOCOL_PORT
+
+        // Arrange
+        String portValue = "8000";
+        Map<String, String> additionalProperties = new HashMap<>();
+        additionalProperties.put(NiFiProperties.CLUSTER_NODE_PROTOCOL_PORT, portValue);
+        NiFiProperties properties = loadNiFiProperties("/NiFiProperties/conf/nifi.blank.properties", additionalProperties);
+
+        // Act
+        Integer clusterProtocolPort = properties.getClusterNodeProtocolPort();
+
+        // Assert
+        assertEquals(Integer.parseInt(portValue), clusterProtocolPort.intValue());
+    }
+
+    @Test(expected = NumberFormatException.class)
+    public void testShouldVerifyExceptionThrownWhenInValidFormatPortValue() {
+        // Testing with CLUSTER_NODE_PROTOCOL_PORT
+
+        // Arrange
+        // Port Value is invalid Format
+        String portValue = "8000";
+        Map<String, String> additionalProperties = new HashMap<>();
+        additionalProperties.put(NiFiProperties.CLUSTER_NODE_PROTOCOL_PORT, portValue);
+        NiFiProperties properties = loadNiFiProperties("/NiFiProperties/conf/nifi.blank.properties", additionalProperties);
+
+        // Act
+        Integer clusterProtocolPort = properties.getClusterNodeProtocolPort();
+
+        // Assert
+        // Expect NumberFormatException thrown
+        assertEquals(Integer.parseInt(portValue), clusterProtocolPort.intValue());
```

Comment:
This unit test is failing. NumberFormatException is not thrown. A value of `8000` should (and does) parse correctly.

Reply:
@markap14 I fixed the test. Thanks.